lib/polyfill/orientation.js

```diff
@@ -23,21 +24,47 @@ shaka.polyfill.Orientation = class {
    * @export
    */
   static install() {
-    if (screen.orientation) {
+    if (screen.orientation && screen.orientation.unlock) {
       // Not needed.
+      shaka.log.info('Using native screen.orientation');
       return;
     }
 
-    // There is no way to check to see if the 'orientationchange' event exists
-    // on window, which could theoretically lead to this making a
-    // screen.orientation object that doesn't actually work.
-    // However, it looks like all platforms that support the deprecated
-    // window.orientation feature also support that event.
-    if (window.orientation != undefined) {
+    if (screen.orientation != undefined) {
+      // There are some platforms where screen.orientation is defined but
+      // incomplete (e.g. Safari).
+      // Install a very simple polyfill in that case.
+      shaka.polyfill.Orientation.installBasedOnScreenMethods_();
+    } else if (window.orientation != undefined) {
+      // There is no way to check to see if the 'orientationchange' event exists
+      // on window, which could theoretically lead to this making a
+      // screen.orientation object that doesn't actually work.
+      // However, it looks like all platforms that support the deprecated
+      // window.orientation feature also support that event.
       shaka.polyfill.Orientation.installBasedOnWindowMethods_();
     }
   }
 
+  /**
+   * Modifies screen.orientation to add no-ops for missing methods.
+   * Meant for cases where screen.orientation is defined, but has missing
+   * methods that cannot be properly polyfilled.
+   * @private
+   */
+  static installBasedOnScreenMethods_() {
+    if (screen.orientation.lock === undefined) {
+      screen.orientation.lock = (orientation) => {
+        shaka.log.info('screen.orientation.lock is a no-op');
```

Comment:
What happens when this no-op is used?  How does the app behavior differ from the normal case?

Reply:
Well, this no-op doesn't change anything because we already wrap the `lock` method in a try-catch to silence errors. I just included it for sake of completeness.